pydecorate/decorator_base.py

```diff
@@ -278,7 +278,15 @@ def _add_text(self, txt, **kwargs):
         # draw base
         px = self.style["propagation"][0] + self.style["newline_propagation"][0]
         py = self.style["propagation"][1] + self.style["newline_propagation"][1]
-        x1 = x + px * (tw + 2 * mx)
+        if self.style["extend"]:
+            if self.style["propagation"][0] == 1:
+                x1 = x + px * (x_size - x)
+            elif self.style["propagation"][0] == -1:
+                x1 = x + px * x
+            else:
+                x1 = x + px * (tw + 2 * mx)
+        else:
+            x1 = x + px * (tw + 2 * mx)
```

Comment:
Could we put this in another helper function or method? And how similar is this code to the scale/colorbar code? Could it be reused?

What do you think about a test that just makes sure that the result is different based on extend False and extend True? It doesn't have to check actual pixel-by-pixel comparison.